### PR TITLE
{containerapp} remove additional conversion of binding name of Java component

### PR DESCRIPTION
**Pull request by ShichaoQiu**
remove additional conversion of binding name of Java component

### CODE REVIEW

**Comment by yonzhan** [comment]
Containerapp

**Review by Greedygre — APPROVED** [review]

**Comment by Greedygre** [comment]
This PR is for fix a logic which has not been released for customer in PR https://github.com/Azure/azure-cli-extensions/pull/7216, so no need to update `history`

File: src/containerapp/azext_containerapp/java_component_decorator.py [hunk 1]
```diff
@@ -57,6 +57,10 @@ def create(self):
                 environment_name=self.get_argument_environment_name(), name=self.get_argument_java_component_name(),
                 java_component_envelope=self.java_component_def, no_wait=self.get_argument_no_wait())
         except Exception as e:
+            stringErr = str(e)
+            if "JavaComponentsNotAllowedForSubscription" in stringErr:
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/java_component_decorator.py:61]: Is it possible to get ErrorCode?

**Review by Juliehzl — COMMENTED** [review]
add test case for '-' scenario

File: src/containerapp/azext_containerapp/java_component_decorator.py (hunk 1 above)
Comment (ShichaoQiu) [reply src/containerapp/azext_containerapp/java_component_decorator.py:61]: The error code is 400. You mean that we need to check both of the error code and the error message, right?

Comment (ShichaoQiu) [reply src/containerapp/azext_containerapp/java_component_decorator.py:61]: As we discussed offline, will close this

**Review by Greedygre — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**